slowquant/qiskit_interface/custom_ansatz.py

```diff
@@ -142,12 +145,15 @@ def fUCC(
     Returns:
         Factorized UCCSD ansatz circuit and R parameters needed for gradients.
     """
-    valid_options = ()
+    valid_options = ("n_layers",)
     for option in ansatz_options:
         if option not in valid_options:
             raise ValueError(f"Got unknown option for fUCC, {option}. Valid options are: {valid_options}")
     if not isinstance(mapper, JordanWignerMapper):
-        raise ValueError(f"efficientUCCSD only implemented for JW mapper, got: {type(mapper)}")
+        raise ValueError(f"fUCC only implemented for JW mapper, got: {type(mapper)}")
+    if "n_layers" not in ansatz_options.keys():
+        raise ValueError("kSAdUpCCGSD require the option 'n_layers'")
```

Comment:
You mean fUCC ?  (copy-paste error?)
Also would a default of n_layers = 1 not be nice? Does this not break code?